stdlib/public/core/IntegerParsing.swift

```diff
@@ -146,31 +45,44 @@ extension FixedWidthInteger {
   ///     `radix`.
   ///   - radix: The radix, or base, to use for converting `text` to an integer
   ///     value. `radix` must be in the range `2...36`. The default is 10.
-  @inlinable // @specializable
-  @_semantics("optimize.sil.specialize.generic.partial.never")
+  @inlinable @inline(__always)
   public init?<S: StringProtocol>(_ text: S, radix: Int = 10) {
-    _precondition(2...36 ~= radix, "Radix not in range 2...36")
-
-    if let str = text as? String, str._guts.isFastUTF8 {
-      guard let ret = str._guts.withFastUTF8 ({ utf8 -> Self? in
-        var iter = utf8.makeIterator()
-        return _parseASCII(codeUnits: &iter, radix: Self(radix))
-      }) else {
-        return nil
-      }
-      self = ret
-      return
+    precondition(2...36 ~= radix)
+    // Check we can load the UTF-8 in UInt64 chunks.
+    let wholeGuts = text._wholeGuts
+    let canChunkLoad: Bool
+    if S.self == String.self {
+      canChunkLoad = wholeGuts.isSmall || wholeGuts.count >= 9 // +1 for sign.
+    } else {
+      canChunkLoad = (wholeGuts.isSmall ? 16 : wholeGuts.count)
+        &- text.startIndex._encodedOffset >= 9
+    }
+    if _fastPath(canChunkLoad), _fastPath(wholeGuts.isFastUTF8),
+        Self.bitWidth <= 64 {
+      let r64_ = wholeGuts.withFastUTF8({ utf8 -> UInt64? in
+        let utf8 = (S.self == String.self) ? utf8 :
+          UnsafeBufferPointer(rebasing: utf8[text._offsetRange])
+        let isSigned = Self.isSigned
+        let max = UInt64(Self.max)
+        switch radix {
+        case 10: return _parseIntegerBase10(from: utf8,
+                                            allowNegative: isSigned, max: max)
+        case 16: return _parseIntegerBase16(from: utf8,
+                                            allowNegative: isSigned, max: max)
+        case 2: return _parseIntegerBase2(from: utf8,
+                                          allowNegative: isSigned, max: max)
+        default: return _parseInteger(from: utf8, radix: radix,
+                                      allowNegative: isSigned, max: max)
+        }
+      })
+      guard _fastPath(r64_ != nil), let r64 = r64_ else { return nil }
+      self = Self(truncatingIfNeeded: r64)
+    } else {
+      var iter = text.utf8.makeIterator()
+      let result_: Self? = _parseInteger(from: &iter, radix: radix)
+      guard _fastPath(result_ != nil), let result = result_ else { return nil }
```

Comment:
Are you seeing any codegen differences from some of the more liberal uses of `_fastPath`?

Reply:
I primarily added them for branch hints, do you have any concern regarding adding these?

Reply:
Oh just noticed this particular one is completely redundant. Must've been left over from some refactoring.